rust/agama-server/src/l10n/model.rs

```diff
@@ -266,21 +173,20 @@ impl L10n {
         Ok(())
     }
 
-    fn x11_keymap() -> Result<String, io::Error> {
-        let output = run_with_timeout(
-            &["setxkbmap", "-query", "-display", &display()],
-            SETXKBMAP_TIMEOUT,
-        );
-        let output = output.map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
-        let output = output.unwrap_or(String::new());
+    fn ui_keymap() -> Result<KeymapId, LocaleError> {
+        let output = Command::new("/usr/bin/localectl")
+            .output()
+            .map_err(LocaleError::Commit)?;
+        let output = String::from_utf8_lossy(&output.stdout);
 
-        let keymap_regexp = Regex::new(r"(?m)^layout: (.+)$").unwrap();
+        let keymap_regexp = Regex::new(r"(?m)VC Kayout: (.+)$").unwrap();
```

Comment:
that regexp looks strange...on my system I have `VC Keymap: us` and on my testing VM it is identical. So unless there is some change of output in wayland ( which I do not test), I would say that this regexp is broken.

Reply:
Yes, it looks broken to me.

Reply:
I have fixed the problem and I did the proper test 🫤 (change the locale, restart the server and check the configuration).